uqcsbot/scripts/crates.py

```diff
@@ -0,0 +1,561 @@
+    cats, tot = get_category_page(channel, args.sort, 1)
+    if cats is None:
+        return  # Error occurred
+
+    # Get all of the categories by incrementing page number
+    page = 2
+    while len(cats) < tot:
+        next_cats, _ = get_category_page(channel, args.sort, page)
+        if next_cats is None or not next_cats:
+            break
+
+        cats.extend(next_cats)
+        page += 1
+
+    # Begin formatting the message
+    category_string = '\n'.join(cats)
+    blocks = [
+        {
+            'type': 'section',
+            'text': {
+                'type': 'mrkdwn',
+                'text': f'*Displaying {tot} categories:*'
+            }
+        },
+        {
+            'type': 'section',
+            'text': {
+                'type': 'mrkdwn',
+                'text': f'```{category_string}```'
+            }
+        },
+    ]
+
+    bot.post_message(channel, '', blocks=blocks)
+
+
+def display_specific_category(channel: Channel, args: CategorySearch):
+    """Displays a single category in more detail"""
+    # Get the categories
+    url = BASE_URL + f'/categories/{args.name}'
+    response = requests.get(url)
+
+    if response.status_code != requests.codes.ok:
+        bot.post_message(channel, f'There was a problem getting the category "{args.name}"')
+        return
+
+    # Convert the json response
+    response_data = json.loads(response.content)
+    if 'errors' in response_data:
+        bot.post_message(channel, f'The category "{args.name}" does not exist')
+        return
+
+    raw_category = response_data.get('category')
+
+    name = raw_category.get('name')
+    name = raw_category.get('id') if name is None else name
+    desc = raw_category.get('description', 'No description provided')
+
+    category = CategoryResult(name, desc, raw_category['crates_cnt'])
+
+    # Format the message
+    blocks = [
+        {
+            'type': 'section',
+            'text': {
+                'type': 'mrkdwn',
+                'text': f'*{category.name}:*'
+            },
+        },
+        {
+            'type': 'section',
+            'text': {
+                'type': 'mrkdwn',
+                'text': f'{category.description}'
+            },
+        },
+        {
+            'type': 'context',
+            'elements': [
+                {
+                    'type': 'plain_text',
+                    'text': f'Crate Count: {category.crates}'
+                }
+            ]
+        },
+    ]
+
+    bot.post_message(channel, '', blocks=blocks)
+
+
+def handle_categories_route(channel: Channel, args: CategorySearch):
+    """Handles the categories sub-command by determining whether or not to display all categories or just one"""
+    if args.name:
+        display_specific_category(channel, args)
+    else:
+        display_all_categories(channel, args)
+
+
+def get_user(channel: Channel, username: str) -> Optional[UserResult]:
```

Comment:
I don't know if it's just me, but it took a while to work out that this was referring to crates.io users, not slack ones. Could it be made more clear?